x-pack/legacy/plugins/infra/public/components/metrics/node_details.tsx

```diff
@@ -0,0 +1,153 @@
+      defaultMessage: 'Instance ID',
+    }),
+  },
+  {
+    field: 'cloud.provider',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.cloudProvider', {
+      defaultMessage: 'Cloud Provider',
+    }),
+  },
+  {
+    field: 'host.os.name',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.operatinSystem', {
+      defaultMessage: 'Operating System',
+    }),
+  },
+  {
+    field: 'host.os.kernel',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.kernelVersion', {
+      defaultMessage: 'Kernel Version',
+    }),
+  },
+  {
+    field: 'host.hostname',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.hostname', {
+      defaultMessage: 'Hostname',
+    }),
+  },
+  {
+    field: 'host.containerized',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.containerized', {
+      defaultMessage: 'Containerized',
+    }),
+    isBoolean: true,
+  },
+  {
+    field: 'cloud.project.id',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.projectId', {
+      defaultMessage: 'Project ID',
+    }),
+  },
+  {
+    field: 'cloud.availability_zone',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.availabilityZone', {
+      defaultMessage: 'Availability Zone',
+    }),
+  },
+  {
+    field: 'cloud.machine.type',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.machineType', {
+      defaultMessage: 'Machine Type',
+    }),
+  },
+  {
+    field: 'cloud.instance.name',
+    label: i18n.translate('xpack.infra.nodeDetails.labels.instanceName', {
+      defaultMessage: 'Instance Name',
+    }),
+  },
+] as FieldDef[];
+
+const getLabelForField = ({ field }: FieldDef) => {
+  const fieldDef = FIELDS.find(f => f.field === field);
+  if (!fieldDef) return field;
+  return fieldDef.label;
+};
+
+const getValueForField = (metadata: InfraMetadata, { field, isBoolean }: FieldDef) => {
+  if (isBoolean) {
+    return get(metadata.info, field, false)
+      ? i18n.translate('xpack.infra.nodeDetails.yes', { defaultMessage: 'Yes' })
+      : i18n.translate('xpack.infra.nodeDetails.no', { defaultMessage: 'No' });
+  }
+  const value = get(metadata.info, field, '--');
+  return value;
+};
+
+export const NodeDetails = ({ metadata }: Props) => {
+  const [isOpen, setControlState] = useState<boolean>(false);
+
+  const handleClick = useCallback(() => (isOpen ? setControlState(false) : setControlState(true)), [
+    isOpen,
+  ]);
+
+  if (!metadata) {
+    return null;
+  }
+
+  return (
+    <NodeDetailsContainer>
+      <Controls>
+        <EuiButtonIcon
+          iconType={isOpen ? 'arrowUp' : 'arrowDown'}
+          onClick={handleClick}
+          aria-label={i18n.translate('xpack.infra.nodeDetails.labels.showMoreDetails', {
+            defaultMessage: 'Show more details',
+          })}
+        />
+      </Controls>
+      <EuiFlexGrid columns={4} style={{ flexGrow: 1 }}>
+        {FIELDS.slice(0, isOpen ? FIELDS.length : 4).map(field => (
```

Comment:
Not sure if this matters or not, but this is a wasted slice. Could just `useMemo` and create `getFields` so you'd only need to run the `.slice` at most 2 times.